Remove incremental attribute and search event

annevk opened the pull request:
#### 99ff08340f12db8d867a0ed04f176802b7e0a355
<pre>
Remove incremental attribute and search event
<a href="https://bugs.webkit.org/show_bug.cgi?id=195818">https://bugs.webkit.org/show_bug.cgi?id=195818</a>
rdar://48937114

Reviewed by Aditya Keerthi.

These are non-standard and barely used. They are put behind a flag and
will be kept around for a while in case they are still being used
somewhere.

* LayoutTests/TestExpectations:
* LayoutTests/fast/dom/Window/get-set-properties-expected.txt:
* LayoutTests/fast/dom/Window/get-set-properties.html:
* LayoutTests/fast/dom/Window/resources/window-properties.js:
* LayoutTests/fast/dom/Window/window-lookup-precedence-expected.txt:
* LayoutTests/fast/dom/event-attribute-availability-expected.txt:
* LayoutTests/fast/dom/event-attribute-availability.html:
* LayoutTests/fast/dom/event-handler-attributes-expected.txt:
* LayoutTests/fast/dom/event-handler-attributes.html:
* LayoutTests/fast/dom/resources/wrapper-identity-base.js:
* LayoutTests/fast/dom/wrapper-identity-expected.txt:
* LayoutTests/fast/events/onsearch-enter-expected.txt: Removed.
* LayoutTests/fast/events/onsearch-enter.html: Removed.
* LayoutTests/fast/forms/incremental-dom-property-expected.txt: Removed.
* LayoutTests/fast/forms/incremental-dom-property.html: Removed.
* LayoutTests/fast/forms/input-search-press-escape-key-expected.txt: Removed.
* LayoutTests/fast/forms/input-search-press-escape-key.html: Removed.
* LayoutTests/fast/forms/search-event-delay-expected.txt: Removed.
* LayoutTests/fast/forms/search-event-delay.html: Removed.
* LayoutTests/http/tests/security/cross-frame-access-get-expected.txt:
* LayoutTests/http/tests/security/cross-frame-access-get.html:
* LayoutTests/http/tests/security/cross-frame-access-getOwnPropertyDescriptor-expected.txt:
* LayoutTests/http/tests/security/cross-frame-access-getOwnPropertyDescriptor.html:
* LayoutTests/http/tests/security/cross-frame-access-put-expected.txt:
* LayoutTests/http/tests/security/cross-frame-access-put.html:
* LayoutTests/http/tests/security/resources/cross-frame-iframe-for-put-test.html:
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/forms/historical-expected.txt:
* LayoutTests/platform/gtk/fast/dom/Window/window-lookup-precedence-expected.txt:
* LayoutTests/platform/ios/TestExpectations:
* LayoutTests/platform/mac-wk2/fast/dom/Window/window-lookup-precedence-expected.txt:
* LayoutTests/platform/mac/fast/dom/objc-wrapper-identity-expected.txt:
* LayoutTests/platform/wincairo/TestExpectations:
* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WTF/wtf/cocoa/RuntimeApplicationChecksCocoa.h:
* Source/WebCore/bindings/js/WebCoreBuiltinNames.h:
* Source/WebCore/dom/GlobalEventHandlers.idl:
* Source/WebCore/html/HTMLInputElement.cpp:
(WebCore::HTMLInputElement::defaultEventHandler):
* Source/WebCore/html/HTMLInputElement.idl:
* Source/WebCore/html/SearchInputType.cpp:
(WebCore::SearchInputType::handleKeydownEvent):
(WebCore::SearchInputType::searchEventsShouldBeDispatched const):
* Source/WebCore/html/shadow/TextControlInnerElements.cpp:
(WebCore::SearchFieldCancelButtonElement::defaultEventHandler):
* Source/WebCore/rendering/RenderSearchField.cpp:
(WebCore::RenderSearchField::valueChanged):
* Source/WebKit/Shared/WebPreferencesDefaultValues.cpp:
(WebKit::defaultSearchInputIncrementalAttributeAndSearchEventEnabled):
* Source/WebKit/Shared/WebPreferencesDefaultValues.h:
* Source/WebKitLegacy/mac/WebView/WebPreferencesDefaultValues.h:
* Source/WebKitLegacy/mac/WebView/WebPreferencesDefaultValues.mm:
(WebKit::defaultSearchInputIncrementalAttributeAndSearchEventEnabled):

Canonical link: <a href="https://commits.webkit.org/266068@main">https://commits.webkit.org/266068@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/13e22aaf316f629170c0710c0051ac29c6718827

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [  ~~🧪 style~~](https://ews-build.webkit.org/#/builders/38/builds/12759 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🛠 ios~~](https://ews-build.webkit.org/#/builders/26/builds/13085 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🛠 mac~~](https://ews-build.webkit.org/#/builders/14/builds/13411 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/5/builds/14498 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/12199 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [  ~~🧪 bindings~~](https://ews-build.webkit.org/#/builders/9/builds/12820 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🛠 ios-sim~~](https://ews-build.webkit.org/#/builders/23/builds/15586 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🛠 mac-AS-debug~~](https://ews-build.webkit.org/#/builders/16/builds/13104 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/5/builds/14498 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | 
| [  ~~🧪 webkitperl~~](https://ews-build.webkit.org/#/builders/11/builds/12923 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/23/builds/15586 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/14/builds/13411 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/2/builds/14944 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/23/builds/15586 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/14/builds/13411 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/2/builds/14944 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | 
| [  ~~🛠 🧪 jsc~~](https://ews-build.webkit.org/#/builders/20/builds/10851 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/23/builds/15586 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/14/builds/13411 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/2/builds/14944 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | 
| [  ~~🛠 🧪 jsc-arm64~~](https://ews-build.webkit.org/#/builders/12/builds/12080 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/7/builds/12176 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/16/builds/13104 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🛠 jsc-armv7~~](https://ews-build.webkit.org/#/builders/35/builds/12814 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/30/builds/11433 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/14/builds/13411 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 jsc-armv7-tests~~](https://ews-build.webkit.org/#/builders/35/builds/12814 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | 
| [  ~~🛠 🧪 merge~~](https://ews-build.webkit.org/#/builders/19/builds/3127 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/4/builds/15748 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [  ~~🛠 jsc-mips~~](https://ews-build.webkit.org/#/builders/24/builds/13177 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/31/builds/12019 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [  ~~🧪 jsc-mips-tests~~](https://ews-build.webkit.org/#/builders/24/builds/13177 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | 
<!--EWS-Status-Bubble-End-->

annevk commented on Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
```diff
@@ -5677,6 +5677,20 @@ ScrollingPerformanceTestingEnabled:
+  status: embedder
+  category: html
+  humanReadableName: "Search control incremental attribute and search event"
+  humanReadableDescription: "Disable search control incremental attribute and search event support"
```
```suggestion
  humanReadableDescription: "Support search control incremental attribute and search event support"
```

pxlcoder commented on Source/WTF/wtf/cocoa/RuntimeApplicationChecksCocoa.h:
```diff
@@ -86,6 +86,7 @@ enum class SDKAlignedBehavior {
     RestrictsBaseURLSchemes,
     RunningBoardThrottling,
     ScrollViewContentInsetsAreNotObscuringInsets,
+    SearchInputIncrementalAttributeAndSearchEventDisabled,
```
Maybe we should call this `NoSearchInputIncrementalAttributeAndSearchEvent` for consistency with the other removed API behaviors?

annevk committed:
Remove incremental attribute and search event
https://bugs.webkit.org/show_bug.cgi?id=195818
rdar://48937114

Reviewed by Aditya Keerthi.

These are non-standard and barely used. They are put behind a flag and
will be kept around for a while in case they are still being used
somewhere.

* LayoutTests/TestExpectations:
* LayoutTests/fast/dom/Window/get-set-properties-expected.txt:
* LayoutTests/fast/dom/Window/get-set-properties.html:
* LayoutTests/fast/dom/Window/resources/window-properties.js:
* LayoutTests/fast/dom/Window/window-lookup-precedence-expected.txt:
* LayoutTests/fast/dom/event-attribute-availability-expected.txt:
* LayoutTests/fast/dom/event-attribute-availability.html:
* LayoutTests/fast/dom/event-handler-attributes-expected.txt:
* LayoutTests/fast/dom/event-handler-attributes.html:
* LayoutTests/fast/dom/resources/wrapper-identity-base.js:
* LayoutTests/fast/dom/wrapper-identity-expected.txt:
* LayoutTests/fast/events/onsearch-enter-expected.txt: Removed.
* LayoutTests/fast/events/onsearch-enter.html: Removed.
* LayoutTests/fast/forms/incremental-dom-property-expected.txt: Removed.
* LayoutTests/fast/forms/incremental-dom-property.html: Removed.
* LayoutTests/fast/forms/input-search-press-escape-key-expected.txt: Removed.
* LayoutTests/fast/forms/input-search-press-escape-key.html: Removed.
* LayoutTests/fast/forms/search-event-delay-expected.txt: Removed.
* LayoutTests/fast/forms/search-event-delay.html: Removed.
* LayoutTests/http/tests/security/cross-frame-access-get-expected.txt:
* LayoutTests/http/tests/security/cross-frame-access-get.html:
* LayoutTests/http/tests/security/cross-frame-access-getOwnPropertyDescriptor-expected.txt:
* LayoutTests/http/tests/security/cross-frame-access-getOwnPropertyDescriptor.html:
* LayoutTests/http/tests/security/cross-frame-access-put-expected.txt:
* LayoutTests/http/tests/security/cross-frame-access-put.html:
* LayoutTests/http/tests/security/resources/cross-frame-iframe-for-put-test.html:
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/forms/historical-expected.txt:
* LayoutTests/platform/gtk/fast/dom/Window/window-lookup-precedence-expected.txt:
* LayoutTests/platform/ios/TestExpectations:
* LayoutTests/platform/mac-wk2/fast/dom/Window/window-lookup-precedence-expected.txt:
* LayoutTests/platform/mac/fast/dom/objc-wrapper-identity-expected.txt:
* LayoutTests/platform/wincairo/TestExpectations:
* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WTF/wtf/cocoa/RuntimeApplicationChecksCocoa.h:
* Source/WebCore/bindings/js/WebCoreBuiltinNames.h:
* Source/WebCore/dom/GlobalEventHandlers.idl:
* Source/WebCore/html/HTMLInputElement.cpp:
(WebCore::HTMLInputElement::defaultEventHandler):
* Source/WebCore/html/HTMLInputElement.idl:
* Source/WebCore/html/SearchInputType.cpp:
(WebCore::SearchInputType::handleKeydownEvent):
(WebCore::SearchInputType::searchEventsShouldBeDispatched const):
* Source/WebCore/html/shadow/TextControlInnerElements.cpp:
(WebCore::SearchFieldCancelButtonElement::defaultEventHandler):
* Source/WebCore/rendering/RenderSearchField.cpp:
(WebCore::RenderSearchField::valueChanged):
* Source/WebKit/Shared/WebPreferencesDefaultValues.cpp:
(WebKit::defaultSearchInputIncrementalAttributeAndSearchEventEnabled):
* Source/WebKit/Shared/WebPreferencesDefaultValues.h:
* Source/WebKitLegacy/mac/WebView/WebPreferencesDefaultValues.h:
* Source/WebKitLegacy/mac/WebView/WebPreferencesDefaultValues.mm:
(WebKit::defaultSearchInputIncrementalAttributeAndSearchEventEnabled):

Canonical link: https://commits.webkit.org/266068@main
